src/platform/packages/shared/kbn-esql-ast/src/parser/formatting.ts

```diff
@@ -183,6 +184,32 @@ const attachCommentDecoration = (
     return;
   }
 
+  const commentLocation = comment.node.location;
+
+  // Check if comment belongs to a subquery (parens expression)
+  // Walk through the AST to find parens nodes that contain this comment
+  let commentAttachedToSubquery = false;
+  walk(ast, {
+    visitParens(node) {
+      if (
+        node.child?.type === 'query' &&
+        node.location &&
+        commentLocation.min >= node.location.min &&
+        commentLocation.max <= node.location.max
+      ) {
+        // This comment is inside a subquery, attach it to the child query instead
+        const childQuery = node.child as ESQLAstQueryExpression;
+        attachCommentDecoration(childQuery, tokens, comment);
```

Comment:
We should not need the `Walker` here and we should not call `attachCommentDecoration` recursively within itself.

One place to check is that `Visitor` is working correctly. Specifically `Visitor.findNodtAtOr{After,Before}` mehtods should support `parens` nodes.